client/signup/steps/survey/survey-step-v2.jsx

```diff
@@ -107,7 +107,7 @@ export default React.createClass( {
 
 	handleOtherWriteIn( e ) {
 		this.setState( {
-			otherWriteIn: e.target.value
+			otherWriteIn: e.target.value.replace( /^(\s|\.|\,|\:|\!|\?)+|(\s|\.|\,|\:|\!|\?)+$/g, '' ),
```

Comment:
How about `.replace( /^([^\w]|\s)+|([^\w]|\s)+$/g, '' )` to replace all non-alphanumeric characters and underscores?